### PR TITLE
Added knn model scropt

### PR DESCRIPTION
**Pull request by walterwhiteuhh**
added knn_model to scripts

### CODE REVIEW

File: src/knn_model.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    )
+
+# set a random seed
+RSEED = 42
+np.random.seed(RSEED)
+
+# load the base data from the CSV files
+dataset_name = 'Train_Dataset4'
+df = pd.read_csv(f"{OUTPUT_DIR}/{dataset_name}.csv")
+df
```
Comment (Fbehr-data) [inline src/knn_model.py:68]: add df_test

File: src/knn_model.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+# do the train-test-split
+df_train, df_test = train_test_split_fields(
```
Comment (Fbehr-data) [inline src/knn_model.py:71]: Don't split the data. Use the entire data from train

**Review by Fbehr-data — CHANGES_REQUESTED** [review]

File: src/knn_model.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+# import the needed modules
+import numpy as np
+import pandas as pd
+
+# import the machine learning modules
+# The goal of this file will be the modeling based on the KNN 
+from imblearn.over_sampling import SMOTE
+from sklearn.neighbors import KNeighborsClassifier
+from xgboost import XGBClassifier, DMatrix, cv
```
Comment (langer-net) [inline src/knn_model.py:22]: You don't need line 22. Better leave it out.

File: src/knn_model.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+# import own modules from the scr folder
+from train_test_function import train_test_split_fields
+from eda_functions import (
+    plot_confusion_matrix,
+    get_label_accuracies,
+    plot_label_accuracy,
+    plot_feature_importance
+    )
+
```
Comment (langer-net) [inline src/knn_model.py:61]: Remove lines 53 to 60, you don't need them.

**Comment by walterwhiteuhh** [comment]
made suggested changes on knn_model

**Review by Fbehr-data — APPROVED** [review]

**Pull request merged**